extensions/put-io-downloader/package.json

```diff
@@ -0,0 +1,97 @@
+{
+  "$schema": "https://www.raycast.com/schemas/extension.json",
+  "name": "put-io-downloader",
+  "title": "Put.io",
+  "description": "Browse files and transfers on Put.io.",
+  "icon": "putio-icon.png",
+  "author": "EpaL",
+  "categories": [
+    "Media",
+    "Productivity"
+  ],
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "transfers",
+      "title": "Browse Transfers",
+      "subtitle": "Put.io",
+      "description": "Browse your Put.io transfers.",
+      "mode": "view"
+    },
+    {
+      "name": "files",
+      "title": "Browse Files",
+      "subtitle": "Put.io",
+      "description": "Browse your Put.io files and perform configurable actions on them.",
+      "mode": "view"
+    }
+  ],
+  "preferences": [
+    {
+      "name": "putioClientId",
+      "description": "Put.io Client ID",
+      "type": "textfield",
+      "required": true,
+      "title": "Put.io Client ID",
+      "placeholder": "Put.io Client ID"
+    },
+    {
+      "name": "putioOAuthToken",
+      "description": "Put.io OAuth Token",
+      "type": "textfield",
+      "required": true,
+      "title": "Put.io OAuth Token",
+      "placeholder": "Put.io OAuth Token"
+    },
+    {
+      "name": "actionTitle1",
+      "description": "Title of the first action. e.g. 'Download to my local machine'",
+      "type": "textfield",
+      "required": false,
+      "title": "Action Title #1",
+      "placeholder": "Download to my local machine"
+    },
+    {
+      "name": "actionCommand1",
+      "description": "The command to run for this action. Use {0} to insert the file URL. e.g. 'wget {0}'",
```

Comment:
what is the use case for these two commands? It feels kind of…easy for a user to shoot themselves in the foot unintentionally. Couldn't we get away with defining a few different actions (delete/open in editor) and provide a dropdown instead?

Reply:
Put.io is a managed BitTorrent service. Users are using it to grab files from BitTorrent sites and then typically download them to their local machine or server. Typically, a user would create a custom command here to do something with the file; the most common would be to download the file to their local computer. e.g. 'wget {0}'. There is a default option here to open the file in the browser, but these options allow the user to run custom commands (e.g. wget) on the file if they have a more complex download workflow. I've renamed to 'Custom Action #1/#2' and tidied up the placeholders to make it more obvious what these are for.